### PR TITLE
Templates, Fix to a 16:9 aspect ratio.

### PR DESCRIPTION
**Pull request by carlos-kelly**
### Description

This fixes a slide to a 16:9 ratio which fits most TVs and helps unify the presenter display view.

Screen Shot 2019-09-09 at 11 29 16 AM

Template Support:

Screen Shot 2019-09-09 at 2 56 44 PM

### CODE REVIEW

File: examples/js/test-js.js [hunk 1]
```diff
@@ -22,16 +22,32 @@ const cppCodeBlock = `#include <iostream>
...
+      >
+        <Text fontSize={12}>
+          Slide {slideNumber + 1} of {numberOfSlides}
+        </Text>
+        <Box padding={10}>
+          <Image src={formidableLogo} width={100} />
+        </Box>
+      </FlexBox>
+    )}
+  >
```
Comment (carlos-kelly) [inline examples/js/test-js.js]: Components we use inside the template automatically consume the used theme.

File: examples/js/test-js.js [hunk 2]
```diff
@@ -22,16 +22,32 @@ const cppCodeBlock = `#include <iostream>
...
   auto answer = curried_add(7)(8);
   std::cout << answer << std::endl;
-  
+
   return 0;
 }`;
 
 const TestJs = () => (
-  <Deck>
-    <Slide slideNum={1}>
```
Comment (carlos-kelly) [inline examples/js/test-js.js]: We no longer need these as we add them via props based on their order in the tree.

File: examples/js/test-js.js [hunk 3]
```diff
@@ -22,54 +22,74 @@ const cppCodeBlock = `#include <iostream>
...
+  <Deck
+    template={({ numberOfSlides, slideNumber }) => (
+      <FlexBox
+        justifyContent="space-between"
+        position="absolute"
+        bottom={0}
+        width={1}
+      >
+        <Text fontSize={12}>
+          Slide {slideNumber + 1} of {numberOfSlides}
```
Comment (SunburtReynolds) [inline examples/js/test-js.js]: If we want the slide # to match the url, then it should be 0 indexed. Might be confusing if they change the url manually

File: examples/js/test-js.js [hunk 4]
```diff
@@ -22,54 +22,74 @@ const cppCodeBlock = `#include <iostream>
...
+    )}
+  >
+    <Slide>
       <Heading>Spectacle</Heading>
       <Text>Hello There 🤗</Text>
       <Quote>This is a Formidaquote!</Quote>
       <OrderedList>
-        <ListItem>This is an</ListItem>
-        <ListItem>ordered list</ListItem>
+        <Appear elementNum={1}>
```
Comment (SunburtReynolds) [inline examples/js/test-js.js]: FYI, elem num should be 0 indexed as well.

File: examples/js/test-js.js (hunk 3 above)
Comment (carlos-kelly) [reply examples/js/test-js.js]: Good point!

File: src/components/slide.js [hunk 5]
```diff
@@ -5,14 +5,27 @@ import { DeckContext } from '../hooks/use-deck';
 import styled from 'styled-components';
 import { color } from 'styled-system';
 
+const SlideContainer = styled('div')`
+  ${color};
+  position: relative;
+  width: 100%;
+  padding-top: 56.25%;
```
Comment (SunburtReynolds) [inline src/components/slide.js]: Nice.

**Pull request merged**